mm2src/coins/solana.rs

```diff
@@ -377,6 +426,311 @@ impl SolanaCoin {
         let guard = self.spl_tokens_infos.lock().unwrap();
         (*guard).clone()
     }
+
+    fn send_hash_time_locked_payment(&self, args: SendPaymentArgs<'_>) -> SolTxFut {
+        let receiver = Pubkey::new(args.other_pubkey.iter().as_slice());
+        let swap_program_id = Pubkey::new(args.swap_contract_address.as_ref().unwrap().as_slice());
+        let amount = sol_to_lamports(args.amount.to_f64().unwrap());
+        let secret_hash: [u8; 32] = <[u8; 32]>::try_from(args.secret_hash).expect("unable to convert to 32 byte array");
```

Comment:
> You can add a checklist in the original issue https://github.com/KomodoPlatform/komodo-defi-framework/issues/1085 here and add an item to research if a better secret hash algorithm can be used for solana, we aim to use the secret hash algorithm that uses the least gas while being compatible with other coins.

You didn't add this item to a checklist as advised.

Also for this https://github.com/KomodoPlatform/komodo-defi-framework/blob/ac8cc07536751d8145949f0542380bcb768ba45c/mm2src/coins/solana.rs#L420-L429
it's better to wrap the block of code in a future like how it's done in some places in `eth.rs` and other files and propagate errors using ? to avoid these long match statements that makes the code less readable.

Reply:
<img width="902" alt="Screenshot 2024-04-02 at 9 57 46 PM" src="https://github.com/KomodoPlatform/komodo-defi-framework/assets/102911930/636086f6-c870-4860-b5ba-f0fd066a77ac">
I have added to the bottom

Reply:
> > You can add a checklist in the original issue #1085 here and add an item to research if a better secret hash algorithm can be used for solana, we aim to use the secret hash algorithm that uses the least gas while being compatible with other coins.
> 
> You didn't add this item to a checklist as advised.
> 
> Also for this
> 
> https://github.com/KomodoPlatform/komodo-defi-framework/blob/ac8cc07536751d8145949f0542380bcb768ba45c/mm2src/coins/solana.rs#L420-L429
> 
> 
> it's better to wrap the block of code in a future like how it's done in some places in `eth.rs` and other files and propagate errors using ? to avoid these long match statements that makes the code less readable.

done